controllers/azuremachine_controller.go

```diff
@@ -272,29 +281,35 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 	switch vm.State {
 	case infrav1.VMStateSucceeded:
 		machineScope.V(2).Info("VM is running", "id", *machineScope.GetVMID())
+		conditions.MarkTrue(machineScope.AzureMachine, infrav1.VMRunningCondition)
 		machineScope.SetReady()
 	case infrav1.VMStateCreating:
 		machineScope.V(2).Info("VM is creating", "id", *machineScope.GetVMID())
+		conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMRunningCondition, infrav1.VMNCreatingReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
side note: right now we never hit these because we block on long running operations such as VM creation so that will need to change in order for conditions to really become useful

Reply:
not feedback for this PR, more of a general comment

Reply:
yes, you are correct, it will skip some of the conditions, but I thought we should add them here anyways so its there when we improve that